rflx/generator/generator.py

```diff
@@ -2301,14 +2301,16 @@ def __create_message_unit(self, message: Message) -> None:
             WithClause(self.__prefix * const.TYPES_PACKAGE),
         ]
 
-        arrays = [
-            self.__prefix * ID(t.identifier) for t in message.types.values() if isinstance(t, Array)
-        ]
+        arrays = {
+            self.__prefix * ID(t.identifier): None
+            for t in message.types.values()
+            if isinstance(t, Array)
+        }
         context.extend(WithClause(array) for array in arrays)
         instantiation = GenericPackageInstantiation(
             self.__prefix * ID(message.identifier),
             name,
-            [self.__prefix * const.TYPES_PACKAGE] + arrays,
+            [self.__prefix * const.TYPES_PACKAGE] + list(arrays.keys()),
```

Comment:
A regression test should be added. Checking the compilability of the reproducer in #536 should be sufficient. `tests/integration/specification_model_generator_test.py` would be an appropriate place for this test.

Reply:
Yes, we should definitely have that. Added in 50eeadd2e01fda7af9e4f9826e1f9e2fa60dc25a.